src/components/Resource/ResourceFilter.tsx

```diff
@@ -40,51 +42,59 @@ export default function ListFilter(props: any) {
     status: filter.status || null,
   });
 
-  const { loading: orginFacilityLoading } = useTanStackQueryInstead(
-    routes.getAnyFacility,
-    {
-      prefetch: filter.origin_facility !== undefined,
-      pathParams: { id: filter.origin_facility },
-      onResponse: ({ res, data }) => {
-        if (res && data) {
-          setFilterState({
-            origin_facility_ref: filter.origin_facility === "" ? "" : data,
-          });
-        }
-      },
-    },
-  );
+  const { isLoading: orginFacilityLoading, data: originFacilityData } =
+    useQuery({
+      queryKey: [routes.getAnyFacility.path, filter.origin_facility],
```

Comment:
```suggestion
      queryKey: ["getAnyFacility", filter.origin_facility],
```

Same goes for ones below 👍